exercises/brandon_hewer/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,216 @@
+#include "main.h"
+
+using namespace std;
+
+/*
+ * Start of execution.
+ */
+int main(int num_args, char ** args) {
+
+	// Check whether an argument has been passed (1 default argument
+	// is always passed; working directory).
+	if (num_args == 1) {
+		cerr << "No file path was given as an argument. Program Exiting..." << endl;
+		return 0;
+	}
+	else {
+		ifstream ifile(args[1]);
+
+		// Check whether a file exists at the first given argument.
+		if (!ifile) {
+			cerr << "No file exists at the specified path. Program Exiting..." << endl;
+			return 0;
+		}
+
+		unordered_set<char> delims = { '.', ',', '?', '!', '(', ')', ':', '\'', 
+			'"', ' ', ';', '\t'};
+		auto word_cond = [](const string s) -> const bool {
+			return s.size() >= MIN_WORD_LENGTH;
+		};
+
+		// Retrieve words from input file.
+		vector<string> words = getWords(ifile, delims);
+		ifile.close();
+
+		// Count the number of occurrences of each unique word.
+		unordered_map<string, int> count = countUniqueWords(words, word_cond);
+
+		// Output the total occurrences of each unique word in a results file.
+		ofstream ofile("results.txt");
+		ofile << "Word\tUsage\n" << flip_map(count);
+		ofile.close();
+
+		return 1;
+	}
+}
+
+/*
+ * Adds the data contained in the specified multimap \p mp, to the
+ * specified output stream \p os.
+ *
+ * Each key-value entry in the multimap is added to the output stream
+ * in the format "KEY\tVALUE\n". Key-value entries are added in reverse
+ * order.
+ *
+ * \param os	The output stream to add to.
+ * \param mp	The multimap whose data to add to the output stream.
+ *
+ * \returns		The output stream, with the data from the output stream
+ *              added.
+ */
+ostream &operator<<(ostream &os, const multimap<int, string>& mp) {
+	multimap<int, string> mmp;
```

Comment:
Use variable names which describe what the variable refers to (not the data structure). Also try to avoid abbreviations (unless for things like `i` as a loop counter and so on which are typical use cases).